### ISSUE TITLE
Missing Some Features

### ISSUE THREAD

**Issue by nvandendries1984** [rainlab/user-plugin#567]
I miss the following features:

1. When a user registers, I would like the activation to be done by email and/or by the administrator.
   At the moment, when a user registers, it is automatically activated. I would like activation to occur only after confirming the email address.
2. Excluding email addresses based on domain name.

I have been looking but can't find it. It would be nice if this features could be added to the backend settings model.

**Comment by chrisvidal** [comment]
it seems that some previous existing features have been parked.
I found a `trait HasLegacyApi` that provide the old user registration logic in there. Maybe that could help

**Comment by chrisvidal** [comment]
I did some digging around and you can actually do check the user email.
The logic is now different. The user can register and access its account data even if its email is not yet verified.
You can test all that logic using the Vanilla theme.
in the theme, you can register and see your profile account, in there you can see a button `verify my email` which run the ajax method onVerifyEmail() from the `account` component

`/**
     * onVerifyEmail sends the verification email to the user
     */
    public function onVerifyEmail()
    {
        $this->actionVerifyEmail();

        if ($flash = Cms::flashFromPost(__("Please check your email for instructions."))) {
            Flash::success($flash);
        }

        $this->page['showLinkSent'] = true;
    }`

based on that we can do it on our own logic, catching the `user/register` event, send this verification email and give access to his account only when the email is verified, if we want to mimic the previous plugin v2 behaviors.

**Comment by daftspunk** [comment]
OK. This came up (again) as a common requirement. An admin approval workflow has been added in v3.7.0

> Excluding email addresses based on domain name.

This could probably be implemented as a plugin or in the registration logic.
